### PR TITLE
Add fyne.TextTruncateMiddle truncation mode

### LINKED ISSUE #6353: Support for fyne.TextTruncateMiddle
### Checklist

- [x] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [x] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

Sometimes it is much more informative to see the beginning and end of a string rather than only the beginning, which `fyne.TextTruncateMiddle` could solve nicely.

### Is it possible to construct a solution with the existing API?

No, `fyne.TextTruncation` does not offer `fyne.TextTruncateMiddle`.

### Describe the solution you'd like to see.

Adding an option and a code-path to handle the rendering should do it.

### PR DESCRIPTION
**Pull request by mohsenm4**
## Description

Adds `fyne.TextTruncateMiddle` alongside the existing `Off`/`Clip`/`Ellipsis` modes. When the text overflows, it shortens with an ellipsis in the middle while preserving the beginning and end — handy for file paths, URLs, etc. where both ends carry information.

```go
label.Truncation = fyne.TextTruncateMiddle
// "/home/user/projects/fyne/widget/label.go"  →  "/home/user/…/widget/label.go"
```

Closes #6353

## How

- New `TextTruncateMiddle` constant in `text.go`.
- `middleTruncate` helper in `widget/richtext.go` — binary search for the largest balanced prefix+ellipsis+suffix that fits the available width.
- New `displayText` field on `rowBoundary` so the renderer can substitute a pre-computed string when the slice-based path doesn't fit the use case. MinSize matches the ellipsis mode.
- Added a fourth option in the fyne_demo truncation radio so it's easy to try live.

## Tests

- `TestText_middleTruncate` — helper edge cases (empty, fits whole, only ellipsis fits, balanced split).
- `TestText_lineBounds_truncateMiddle` — short text untouched, long text gets the ellipsis with original head/tail preserved.
- `TestLabel_TruncateMiddle` — end-to-end on a real Label, asserts the rendered canvas text.

### CODE REVIEW

File: widget/richtext.go [hunk 1]
```diff
@@ -1334,4 +1385,8 @@ type rowBoundary struct {
 	begin, end        int
 	ellipsis          bool
 	indent            float32
+	// displayText overrides the [begin:end] slice when rendering. It is used
+	// to substitute a pre-computed truncated string such as the prefix+…+suffix
+	// produced by [fyne.TextTruncateMiddle].
```
Comment (andydotxyz) [inline widget/richtext.go]: Please don't put "where it is used" documentation into a field - documents work for purpose but the code itself expresses usage (and will vary over time)

File: widget/richtext.go [hunk 2]
```diff
@@ -1276,11 +1287,51 @@ func splitLines(seg RichTextSegment) []rowBoundary {
...
-	return append(lines, rowBoundary{[]RichTextSegment{seg}, len(lines), low, length, false, 0})
+	return append(lines, rowBoundary{[]RichTextSegment{seg}, len(lines), low, length, false, 0, ""})
+}
+
+// middleTruncate finds the largest prefix and suffix rune counts such that
+// runes[:prefix] + "…" + runes[len(runes)-suffix:] fits within maxWidth when
+// measured. The third return value is true when the entire input fits without
+// truncation. The prefix is the larger half when the total is odd, so a single
+// character input is always kept as-is.
+func middleTruncate(runes []rune, maxWidth float32, measurer func([]rune) fyne.Size) (int, int, bool) {
```
Comment (andydotxyz) [inline widget/richtext.go]: I think "truncateMiddle" is a more natural name and fits with the "truncateLimit" below.

Comment (andydotxyz) [reply widget/richtext.go]: The last sentence of the document is confusing, why does the odd length relate to a single character? I don't think this is needed but if it is perhaps "when un-even the prefix will be larger than suffix"?

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Thanks for this, just some notes on naming / docs

File: widget/label_test.go [hunk 3]
```diff
@@ -199,6 +199,31 @@ func TestLabel_ChangeTruncate(t *testing.T) {
...
+	c.SetPadded(false)
+	original := "/home/user/projects/fyne/widget/label.go"
+	label := NewLabel(original)
+	c.SetContent(label)
+	natural := label.MinSize()
+	c.Resize(natural)
+
+	truncSize := fyne.NewSize(natural.Width/2, natural.Height)
+	label.Resize(truncSize)
+	label.Truncation = fyne.TextTruncateMiddle
```
Comment (sdassow) [inline widget/label_test.go:215]: I think if you move the Truncation assignment before the `Resize()` call you can save the `Refresh()`.
